src/content/en/updates/2019/12/webassembly.md

```diff
@@ -0,0 +1,84 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Step over code, set breakpoints, and resolve stack traces in your source languages from within DevTools.
+
+{# wf_updated_on: 2019-12-05 #}
+{# wf_published_on: 2019-12-05 #}
+{# wf_tags: webassembly #}
+{# wf_featured_image: /web/updates/images/generic/chrome-devtools.png #}
+{# wf_featured_snippet: Step over code, set breakpoints, and resolve stack traces in your source languages from within DevTools. #}
+{# wf_blink_components: Platform>DevTools #}
+
+# Improved WebAssembly debugging in Chrome DevTools {: .page-title }
+
+{% include "web/_shared/contributors/ingvarstepanyan.html" %}
+
+## Background
+
+Until recently, the only WebAssembly debugging that Chrome DevTools supported was viewing 
+raw WebAssembly stack traces, and stepping over individual instructions in a 
+disassembled WebAssembly text format.
+
+<figure>
+  <img src="../../images/2019/12/raw.png"
+       alt="A screenshot of the previously limited WebAssembly debugging support in 
+            Chrome DevTools."/>
+</figure>
+
+While this works with any WebAssembly module and helps somewhat with debugging small, isolated functions, it’s not very practical for larger apps where the mapping between the disassembled code and your sources is less obvious.
+
+### A temporary workaround
+
+To work around this problem, Emscripten and DevTools have temporarily adapted the existing 
+[source maps](https://www.html5rocks.com/en/tutorials/developertools/sourcemaps/) format to WebAssembly. This allowed mappings
+between binary offsets in the compiled module to original locations in source files.
+
+<figure>
+  <img src="../../images/2019/12/sourcemaps.png"
+       alt="A screenshot of the source-maps-powered debugging."/>
+</figure>
+
+However, source maps were designed for text formats with clear mappings to JavaScript concepts and values, not for binary formats like WebAssembly with arbitrary source languages, type systems, and a linear memory. This made the integration hacky, limited, and not widely supported outside Emscripten.
+
+## Enter DWARF
+
+On the other hand, many native languages already have a common debugging format,
+[DWARF](http://dwarfstd.org/), that provides all the necessary information for debuggers to resolve locations, variable names, type layouts, and more.
+
+While there are still some WebAssembly-specific features that need to be added for full compatibility, compilers like Clang and Rust already support emitting DWARF information in WebAssembly modules, which enabled the DevTools team to start integrating DWARF information
```

Comment:
Hmm I'm not sure "integrating DWARF information" makes sense as it's just information and is part of user-supplied Wasm modules, not DevTools. I'd leave this either as in original draft

```suggestion
While there are still some WebAssembly-specific features that need to be added for full compatibility, compilers like Clang and Rust already support emitting DWARF information in WebAssembly modules, which enabled the DevTools team to start using it
```

or maybe say

```suggestion
While there are still some WebAssembly-specific features that need to be added for full compatibility, compilers like Clang and Rust already support emitting DWARF information in WebAssembly modules, which enabled the DevTools team to start integrating DWARF support
```

or something similar.